wlauto/instrumentation/poller/poller.c

```diff
@@ -115,15 +128,24 @@ int main(int argc, char ** argv) {
     action.sa_handler = term;
     sigaction(SIGTERM, &action, NULL);
 
-    //Poll files
+    //Poll files 
+    int bytes_read = 0;
     while (!done) {
         gettimeofday(&current_time, NULL);
         time_float = (double)current_time.tv_sec;
         time_float += ((double)current_time.tv_usec)/1000/1000;
         printf("%f", time_float);
-        for (i = 0; i < (argc - optind); i++) {
-            read(files_to_poll[i], buf, 1024);
-            lseek(files_to_poll[i], 0, SEEK_SET);
+        for (i = 0; i < num_files; i++) {
+            lseek(files_to_poll[i].fd, 0, SEEK_SET);
+            bytes_read = read(files_to_poll[i].fd, buf, 1024);
+
+            if (bytes_read < 0) {
+                fprintf(stderr, "WARNING: Read nothing form \"%s\"\n",
+                        files_to_poll[i].path);
+                printf(",");
+                continue;
+            }
+
             strip(buf);
             printf(",%s", buf);
             buf[0] = '\0'; // "Empty" buffer
```

Comment:
You're right, it's no longer necessary, but it does no harm and may prevent issues if the above code is modified further in the future -- "belts and suspenders".